openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,198 @@
+                icc_user = self._create_icc_user(data, duplicated_username)
+                duplicated_username = True
+                call_counter = call_counter + 1
+
+        return icc_user
+
+    def _create_icc_user(self, data, duplicated_username):
+        """
+        Method that creates a user in the ICC database based int the  current user logged in data.
+        Also takes care of duplicated usernames.
+        """
+        log_details = {
+            "url": settings.ICC_CREATE_USER_API_FUNCTION,
+        }
+        icc_user = {}
+
+        try:
+            user, _ = get_user(email=data.get('user_email'))
+            _data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_CREATE_USER_API_FUNCTION,
+                "users[0][username]": user.username + self._get_random_string(
+                    self.USERNAME_SUFFIX_LENGTH) if duplicated_username else user.username,
+                "users[0][password]": self._get_random_string(settings.ICC_HASH_LENGTH),
+                "users[0][firstname]": user.first_name,
+                "users[0][lastname]": user.last_name,
+                "users[0][email]": user.email,
+                "users[0][auth]": "saml2",
+            }
+
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=_data,
+            )
+            log_details['request_payload'] = _data
+
+        except Exception as error:
+            LOG.error('Failed to create ICC user. Reason: %s', str(error))
+            log_details['request_payload']['wstoken'] = 'icc-api-token'
+            log_details['response'] = {'error': 'Failed to create ICC user. Reason: ' + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+
+            pass
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, 'create_user')
+
+        return icc_user
+
+    def _get_icc_user_from_xml_response(self, response, method_type):
+        """
+        Method that receives a xml response and convert it to json object and returns it.
+        """
+        icc_user = {}
+
+        try:
+            str_response = xmltodict.parse(response.content)
+            json_str_response = json.dumps(str_response)
+            json_obj_response = json.loads(json_str_response)
+
+            if method_type == 'create_user':
+                fields = json_obj_response['RESPONSE']['MULTIPLE']['SINGLE']['KEY']
+            else:
+                fields = json_obj_response['RESPONSE']['SINGLE']['KEY'][0]['MULTIPLE']['SINGLE']['KEY']
+
+            for field in fields:
+                if field['@name'] == 'id':
+                    icc_user['id'] = field['VALUE']
+                if field['@name'] == 'username':
+                    icc_user['username'] = field['VALUE']
+                    break
```

Comment:
Does this method only break the loop if enters here?

Reply:
Please, check this one. @ivanvgh

Reply:
Since is a standard response schema the order is id then username and it should break when the last important field is catched, by the way, fields only exists when a user is found.

Reply:
If the order is "always" the same, why do you need to loop through all the fields? @ivanvgh

Reply:
Because it is a list of dicts.
but i can use this for each id and username field:
`icc_user["id"] = next(field for field in fields if field["@name"]=="id").get("VALUE")`
`icc_user["username"] = next(field for field in fields if field["@name"]=="username").get("VALUE")`

but it will loop twice, then in a single loop is preferable.

Reply:
Got it. @ivanvgh